### PR TITLE
SSE integration (PR#139)

### PR DESCRIPTION
**Pull request by ioioioio**
The changes from [PR#139](https://github.com/lemire/simdjson/pull/139) have been integrated. simdjson should now be supporting SSE4.2.

It has been tested by manually setting the json_parse_ptr to the sse implementation. Everything seems fine. The benchmark showed similar performances to the previous version. However, it would probably need to be tested on a real sse-only machine.

The SIMDJSON_PADDING has not been modified. I'm looking for an intelligent way to modify its value on runtime. However, it does not seem to cause any problem right now.

simdutf8check also generates a lot of warnings about unused functions. Changes will be made.

### CODE REVIEW

**Comment by lemire** [comment]
You rock!

We have failing tests under visual studio (see the appveyor log). The issue is that the x86intrin.h header does not exist under Visual Studio... 

When you know that you have an x86/x64 (non-ARM) processor, then you need to have something like this...

```
#ifdef _MSC_VER
/* Microsoft C/C++-compatible compiler */
#include <intrin.h>
#endif 

#ifndef _MSC_VER
/* Non-Microsoft C/C++-compatible compiler */
#include <x86intrin.h>  // on some recent GCC, this will declare posix_memalign
#endif
```

Or something...


cc @sunnygleason

**Comment by ioioioio** [comment]
I was sure I had seen the five green things before I called for the PR :(

I'll check that tomorrow.

**Comment by lemire** [comment]
@ioioioio That's fine. That's why the PR process works so well... because you always have a chance to check things out before the final merge.

**Comment by lemire** [comment]
Everything passes now. I'll do manual QA soon.

**Comment by lemire** [comment]
No noticeable perf. regression when compiling normally.

**Comment by lemire** [comment]
Tests pass when compiling normally.

File: include/simdjson/stage1_find_marks.h [hunk 1]
```diff
@@ -158,29 +178,114 @@ uint64_t compute_quote_mask<instruction_set::neon>(uint64_t quote_bits) {
...
 #endif
-#endif
+#endif // SIMDJSON_AVOID_CLMUL
 
 #ifdef SIMDJSON_UTF8VALIDATE
-template<instruction_set T>really_inline
-void check_utf8(simd_input<T> in,
-                __m256i &has_error,
-                struct avx_processed_utf_bytes &previous) {
+// some hack to bypass the impossibily to overload the check_utf8() specialized template
```
Comment (ioioioio) [inline include/simdjson/stage1_find_marks.h]: That comment is false. I'll find a better one.

**Comment by lemire** [comment]
Having fun with visual studio.

**Comment by lemire** [comment]
Ok @ioioioio ... this looks good. Please review my commits and merge at your convenience.

**Comment by lemire** [comment]
It might be useful to consider adding a few SIMD ISAs ... such sse42_clmul and neon_crypto.

**Pull request merged**